### PR TITLE
chore(lib/trie): refactor `ClearPrefixLimit`

### PR DESCRIPTION
**Pull request by qdm12**
## Changes

- Pass limit and return values deleted count, do not pass limit as pointer downstream
- Simpler code more appropriate for end goal (return number of deleted values)
- Helps #2229 

## Tests

```
go test ./internal/trie/... ./lib/trie/...
```

## Issues

- #2229 

## Primary Reviewer

### CODE REVIEW

File: lib/trie/trie.go [hunk 1]
```diff
@@ -433,8 +433,6 @@ func (t *Trie) insertInBranch(parentBranch *node.Branch, key, value []byte) (new
 
 	oldParentIndex := parentBranch.Key[commonPrefixLength]
 	remainingOldParentKey := parentBranch.Key[commonPrefixLength+1:]
-
-	parentBranch.SetDirty(true)
```
Comment (timwu20) [inline lib/trie/trie.go:437]: why was this removed?

Comment (qdm12) [reply lib/trie/trie.go]: Sorry git rebase glitch, no longer removed.

File: lib/trie/trie.go [hunk 2]
```diff
@@ -640,19 +640,16 @@ func (t *Trie) ClearPrefixLimit(prefixLE []byte, limit uint32) (deleted uint32,
...
 
-	initialLimit := limit
-	t.root, _, allDeleted = t.clearPrefixLimit(t.root, prefix, &limit)
-	deleted = initialLimit - limit
+	t.root, deleted, allDeleted = t.clearPrefixLimit(t.root, prefix, limit)
 	return deleted, allDeleted
 }
 
 // clearPrefixLimit deletes the keys having the prefix till limit reached and returns updated trie root node,
 // true if any node in the trie got updated, and next bool returns true if there is no keys left with prefix.
```
Comment (noot) [inline lib/trie/trie.go:648]: update comment pls

**Review by noot — APPROVED** [review]

**Review by timwu20 — APPROVED** [review]
